jwst/flatfield/flat_field.py

```diff
@@ -215,6 +215,11 @@ def apply_flat_field(science, flat, inverse=False):
     # Combine the science and flat DQ arrays
     science.dq = np.bitwise_or(science.dq, flat_dq)
 
+    # Find all pixels in the flat that have a DQ value of NON_SCIENCE
+    # add the DO_NOT_USE flag to these pixels. We don't want to use these pixels
+    # in futher steps
+    flag_nonsci = np.bitwise_and(science.dq, dqflags.pixel['NON_SCIENCE']).astype(np.bool)
+    science.dq[flag_nonsci] = np.bitwise_or(science.dq[flag_nonsci], dqflags.pixel['DO_NOT_USE'])
```

Comment:
Adding NON_SCIENCE doesn't seem to have any effect on the issue described in the ticket.  I.e. in the dataset from the ticket, running `calwebb_image3` starting with the `_cal.fits` files as they are resolves the issue with just the changes in `outlier_detection`, without having to rerun `flat_field` on the input data.

So what is the purpose of this section of code?  Currently the NON_SCIENCE area in MIRI data gets set to zero in the resampled `_i2d.fits` output.  I.e., I don't see a problem with what is happening currently.